Applications/Spire/Source/Ui/ColorBox.cpp

```diff
@@ -190,6 +184,17 @@ void ColorBox::on_current(const QColor& current) {
 }
 
 void ColorBox::show_color_picker() {
+  if(!m_color_picker) {
+    m_color_picker = new ColorPicker(m_current, *this);
+    m_color_picker->installEventFilter(this);
+    link(*this, *m_color_picker);
+    m_color_picker_panel = m_color_picker->window();
+    m_color_picker_panel->installEventFilter(this);
+    auto input_boxes = get_input_boxes(*m_color_picker);
+    for(auto input_box : input_boxes) {
```

Comment:
```auto& input_box```

Reply:
Since `input_boxes` stores a list of pointers of input boxes, I don't think `auto` can lead to the unnecessary copy overhead. Furthermore, we don't need to modify the pointer itself. So what is the reason for changing `auto` to `auto&`?

Reply:
Use ```auto``` when you need a copy of the value, for example you want to modify a copy of the value. If you just need to view an existing value then a reference is more constrained than a value.